sdk/core/core/inc/az_url.h

```diff
@@ -4,21 +4,41 @@
 #ifndef AZ_URL_H
 #define AZ_URL_H
 
-#include <az_span.h>
 #include <az_pair.h>
+#include <az_result.h>
+#include <az_span.h>
 
 #include <_az_cfg_prefix.h>
 
+typedef struct {
+  az_span userinfo;
+  az_span host;
+  az_span port;
+} az_url_authority;
+
+typedef struct {
+  az_span authority;
+  az_span path;
+} az_url_hier_part;
+
 /**
  * https://en.wikipedia.org/wiki/URL
+ * https://tools.ietf.org/html/rfc3986#section-3.2
  */
 typedef struct {
-  az_span host;
-  az_span path;
-  az_pair_writer query;
+  az_span scheme;
+  az_url_hier_part hier_part;
```

Comment:
I think this would be MUCH easier to use if you just added the userinfo, host, port, and path fields to this structure instead of having nested fields. Other languages do this.

Reply:
I think the code is going to look like this - https://github.com/Azure/azure-sdk-for-c/pull/182/commits/aaf3f953f97d4dcbec346f796f0c692d5f37d3d1

(I am using it instead of the quick-and-dirty URL parser I wrote earlier today, and now you can see "before and after" code demonstration)